src/payment/strategies/checkoutcom/checkoutcom-google-pay-payment-strategy.ts

```diff
@@ -0,0 +1,109 @@
+        }
+
+        const { payment: { methodId = '' } = {} } = payload;
+        const { paymentMethods: { getPaymentMethodOrThrow } } = this._store.getState();
+
+        const testMode: boolean = !!getPaymentMethodOrThrow(methodId).config.testMode;
+        const { checkoutcomkey } = getPaymentMethodOrThrow(methodId).initializationData;
+
+        const {
+            onError = () => {},
+            onPaymentSelect = () => {},
+        } = this._googlePayOptions;
+
+        return Promise.resolve(this._getPayment())
+            .then(payment => {
+                if (!payment.paymentData.nonce || !payment.paymentData.cardInformation) {
+
+                    return this._googlePayPaymentProcessor.displayWallet()
+                        .then(paymentData => this._paymentInstrumentSelected(paymentData))
+                        .then(() => onPaymentSelect())
+                        .then(() => this._getPayment())
+                        .catch(error => {
+                            if (error.statusCode !== 'CANCELED') {
+                                onError(error);
+                            }
+                        });
+                }
+
+            })
+            .then(() => this._convert_token(testMode, checkoutcomkey))
+            .then(payment => {
+                const paymentData: Payment = payment as Payment;
+
+                return this._store.dispatch(this._orderActionCreator.submitOrder({useStoreCredit: payload.useStoreCredit}, options))
+                    .then(() => this._store.dispatch(this._paymentActionCreator.submitPayment(paymentData)));
+            });
+    }
+
+    async requestCheckoutcomTokenize(testMode: boolean | undefined, checkoutcomKey: string, data = {}): Promise<CheckoutcomToken> {
+
+        let url = '';
+        if (testMode) {
+            url = this.TEST_URL;
+        } else {
+            url = this.LIVE_URL;
+        }
+
+        const response = await fetch(url, {
+            method: 'POST',
+            mode: 'cors',
+            cache: 'no-cache',
+            credentials: 'omit',
+            headers: {
+                'Content-Type': ContentType.Json,
+                Authorization: checkoutcomKey,
+            },
+            redirect: 'follow',
+            referrerPolicy: 'no-referrer',
+            body: JSON.stringify(data),
+        });
+
+        return await response.json();
+    }
+
+    private async _convert_token(testMode: boolean, checkoutcomkey: string): Promise<PaymentMethodData> {
+        const payment = this._getPayment();
+        let token: CheckoutcomGooglePayToken | undefined;
+        try {
+            token = JSON.parse(payment.paymentData.nonce);
+        } catch (err) {
+            throw new InvalidArgumentError('Unable to parse response from GooglePay.');
+        }
+
+        if ( token && token.protocolVersion ) {
```

Comment:
What happens if there's no `token`? Is it okay to just do nothing and assume everything is successful? I'd imagine the rest of the code wouldn't be able to function without the token. So I'd throw an explicit error here and let the caller to handle it.

Reply:
yep, changed it, its was not ok